### ISSUE TITLE
shadow distance fadeout

### ISSUE THREAD

**Issue by RomStef** [UPBGE/upbge#843]
Hi guys ,
I notice the Unreal Engine have a funny setting in the light options. it's a fade out effect of the shadow inside the shadow box. this setting allow the shadow to disappear softly in base of distance.

this is a post with the light panel settings.
https://answers.unrealengine.com/questions/60734/shadows-fading-out-with-increased-camera-distance.html?sort=oldest

**Comment by EndSSgamesStudio** [comment]
yes it is an interesting option, but, this can be done easily in BGE or UPBGE, just put the sun directly on the character and make the parent (Ctrl + P) and select vertices

![ll](https://user-images.githubusercontent.com/35436445/46117420-8ba86b00-c1d7-11e8-9357-101e1389d470.gif)

I understand from your point of view, but for me, this is an unnecessary option for now... 😀

**Comment by BluePrintRandom** [comment]
Cascaded sun shadows is a good option

**Comment by RomStef** [comment]
@EndSSgamesStudio i know how to parent sun to the character. i have already implemented in my character controller logic. The point of this issue is that the margin of shadows is too sharp at the border of the shadow box. And in a first person game look not so good

**Comment by EndSSgamesStudio** [comment]
@RomStef 
I got it wrong!
Sorry!

**Comment by RomStef** [comment]
@EndSSgamesStudio don't worry about that , i don't know all the technical terms , many times i don't know how to  explain the features i want to describe.
